contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+<frontmatter>
+  title: Introduction to Ruby on Rails
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Ruby on Rails
+
+Authors: [Chattoraj Ayush](https://github.com/AyushChatto)
+
+## Ruby on Rails overview
+
+**Ruby on Rails** is a web application framework written in the Ruby programming language. It is an opinionated framework, that was designed with the intention of making programming web applications easier and quicker by reducing the amount of code you write, instead replacing it with what it considers the "right" way of doing things - what they call the "Rails" way, although it does allow you to follow your own conventions (at the cost of speed and the productivity boost it promises). 
+
+The Rails philosophy includes two major principles, namely:-
+1. **Don't Repeat Yourself (DRY)**, 
+and
+2. **Convention over Configuration** 
+
+### The Rails Way 
+
+One of the core tenets of Rails is 'Convention Over Configuration'. This means that much of the decisions regarding the architecture of the app, as well some other key decisions, are made for you by Rails, and the default application developed by Rails will follow this convention - thus saving you the effort needed to come up with the code necessary to facilitate it. This is perhaps most evident in Rails' decision to follow a Model-View-Controller (MVC) architecture by default.
+
+The structure of apps following an MVC framework is as follows:-
+
+<center>
+<img src="mvc.jpeg">
+
+_Figure 1. Model-View-Controller Framework_ <sup>[source](https://medium.com/@matthewmain/rails-request-response-cycle-819e9cd8fa4e)</sup>
+</center>
+
+Let us follow along with the diagram, and trace the control flow as a Rails app responds to a request by the client. This app is a standard "Contacts" app that stores the names of "users" of a software.
+
+#### Router
+
+When Rails receives the URL, it first looks up the URI in the `routes.rb` file, which defines, by default, all the routes in the application. Each valid URI is mapped to a function located inside a Controller (the 'C' in MVC), that is then invoked to provide a response to the request. 
+
+In order to achieve this in the "Rails" way, it has a convenient abstraction called a "Resource". A resource refers to a RESTful resource that can be acted upon with the HTTP verbs (GET, POST, PUT, and DELETE), and can be Created, have its details Read, Updated, and Destroyed (CRUD). In order to declare a resource, you would mention it inside your `routes.rb` in the following manner:- 
+
+```ruby
+resources :users
```

Comment:
Perhaps you might want to add a description to explain that we are adding the `users` resource? As someone that is not familiar with ruby, what is happening here is a little confusing to me 😅

Reply:
Maybe have a follow up of `This is equivalent to manually declaring the 7 resourceful routes... something something`